src/utils/derive_parser.rs

```diff
@@ -0,0 +1,27 @@
+use super::value_parsers::kv_parser;
+use crate::utils::auth_token::AuthToken;
+use clap::{command, value_parser, ArgAction::SetTrue, Parser, Subcommand};
+
+#[derive(Parser)]
+pub struct SentryCLI {
+    #[command(subcommand)]
+    pub command: SentryCLICommand,
+
+    #[arg(global=true, long="header", value_name="KEY:VALUE", value_parser=kv_parser, help="Custom headers that should be attached to all requests{n}in key:value format")]
+    pub headers: Vec<(String, String)>,
+
+    #[arg(global=true, long, value_parser=value_parser!(AuthToken), help="Use the given Sentry auth token")]
+    pub auth_token: Option<AuthToken>,
+
+    #[arg(global=true, ignore_case=true, long, value_parser=["trace", "debug", "info", "warn", "error"], help="Set the log output verbosity")]
+    pub log_level: Option<String>,
+
+    #[arg(global=true, action=SetTrue, visible_alias="silent", long, help="Do not print any output while preserving correct exit code. This flag is currently implemented only for selected subcommands")]
+    pub quiet: bool,
+
+    #[arg(global=true, action=SetTrue, long, hide=true, help="Always return 0 exit code")]
+    pub allow_failure: bool,
```

Comment:
Let's just keep as `pub` then. `pub(crate)` and `pub` are equivalent for the `sentry-cli`, since we only publish it as a binary program, not as a library. With either specifier, the item is available universally in the crate, but cannot be accessed from outside.

Reply:
There would only be a difference between the two if this was a library – then, `pub(crate)` would mean the library can access this code, but external users of the library would be blocked. In this case `pub` would allow external library users to access the code